### PR TITLE
Add handled transition

### PR DESCRIPTION
**Pull request by andiCR**
I added a "handledTransition" property to be able to determine, for each geofence, the transition that was handled on the onTransition event (ENTER or EXIT). Might not be the best way to implement it, but it's working on iOS and Android

### CODE REVIEW

File: src/ios/GeofencePlugin.swift [hunk 1]
```diff
@@ -324,13 +324,16 @@ class GeoNotificationManager : NSObject, CLLocationManagerDelegate {
...
-    func handleTransition(region: CLRegion!) {
+    func handleTransition(region: CLRegion!, type: Int) {
         if let geo = store.findById(region.identifier) {
             if let notification = geo["notification"].asDictionary {
                 notifyAbout(geo)
             }
 
-            NSNotificationCenter.defaultCenter().postNotificationName("handleTransition", object: geo.description)
+            var d = geo.description
+            d = d.substringToIndex(advance(d.startIndex, count(d) - 1)) + ",\"handledTransition\":\(type)}"
```
Comment (tsubik) [inline src/ios/GeofencePlugin.swift]: I think it would be better to just change fetched geofence object transitionType like `geo["transitionType"] = type`

**Comment by tsubik** [comment]
Hey, I like the way @jgcaruso has proposed here #49. Changing transitionType of passed object.

**Comment by andiCR** [comment]
I like it as well @tsubik , I didn't want to modify any existing data but i'm this is way more elegant than my solution. That said, we would have to change the json class on ios to enable modification of json variables.

**Comment by tsubik** [comment]
Ohh it is immutable. Duh

**Comment by tsubik** [comment]
It's better to leave those promises, you can use plugin as you like whether with promise or callback way.

**Comment by johnrobertcobbold** [comment]
What this feature given up on ?

**Comment by tsubik** [comment]
Merged in #97

**Pull request closed without merging**